testing/src/scenario/_consistency_checker.py

```diff
@@ -670,6 +670,31 @@ def check_containers_consistency(
             f"Missing from metadata: {diff}.",
         )
 
+    # If you have check-infos, then they must match the plan.
+    for container in state.containers:
+        plan = container.plan
+        for check in container.check_infos:
+            if check.name not in plan.checks:
+                if plan.checks:
+                    plan_has = f"plan has only checks: {', '.join(plan.checks)}"
+                else:
+                    plan_has = "plan has no checks"
+                errors.append(
+                    f"container {container.name!r} has a check {check.name!r} "
+                    f"but the {plan_has}.",
+                )
+                continue
+            plan_check = plan.checks[check.name]
+            for attr_from_plan in ("level", "startup", "threshold"):
+                if getattr(check, attr_from_plan) != getattr(
+                    plan_check, attr_from_plan
+                ):
+                    errors.append(
```

Comment:
I guess that's the common case.

I do wonder though, what if someone tests charm receiving check infos while the plan is being changed, is it hypothetically possible to get info from the previous plan version?

Reply:
Aside... do we support `Mock()` for these leaf fields?
I could imagine someone not bothering with exact field values in their tests.

Reply:
> I do wonder though, what if someone tests charm receiving check infos while the plan is being changed, is it hypothetically possible to get info from the previous plan version?

We don't support using Scenario to test this. Right now, the expectation is that there is a static input state, and that state includes Pebble. It doesn't have much detail, but expanding this sort of functionality is what https://github.com/canonical/operator/issues/1427 Is intended for.